pkgs/stdenv/generic/get-env.sh.nix

```diff
@@ -0,0 +1,136 @@
+builtins.toFile "get-env.sh" ''
```

Comment:
Why this file?

Isn't easier to just create a `get-env.sh` file and pass it to getEnvSh?

Reply:
The formatting is just right here to get the exact same hash that `nix develop` produces. I tried moving it out but ran into hash mismatches, so just left it as is.

Reply:
I agree that we should just have it be a `get-env.sh` file, especially if the point of this PR is to move the `nix-shell`/`nix develop` code into nixpkgs as you [suggested here](https://github.com/NixOS/nixpkgs/pull/206915#discussion_r1052693116).